suite/src/main.rs

```diff
@@ -0,0 +1,30 @@
+#![no_std]
+#![no_main]
+#![feature(custom_test_frameworks)]
+#![feature(default_alloc_error_handler)]
+#![test_runner(crate::runtime::test_runner)]
+#![reexport_test_harness_main = "test_main"]
+
+mod modules;
+mod platform;
+#[macro_use]
+mod runtime;
+
+use platform::Platform;
+use riscv_rt::entry;
+
+extern crate alloc;
+
+fn main() {}
```

Comment:
Could you maybe include a small documentation for every function, trait, module, struct etc that you create?